pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -509,6 +508,46 @@ func (r *ReconcileNetworkAddonsConfig) stopTrackingObjects() {
 	r.statusManager.SetFromPods()
 }
 
+// updateClusterInfo checks specific params in cluster-info struct and updates them.
+func (r *ReconcileNetworkAddonsConfig) updateClusterInfo() error {
+	if r.clusterInfo.OpenShift4 {
+		isSingleReplica, err := isOpenshiftSingleReplica(r.client)
+		if err != nil {
+			log.Printf("failed to check if running on a singleReplica infrastrcuture: %v", err)
+		}
+		r.clusterInfo.IsSingleReplica = isSingleReplica
+	}
+
+	clientset, err := getClientInterface()
+	if err != nil {
+		return fmt.Errorf("failed to get clientset: %v", err)
+	}
+	netAttachDefAvailable, err := isNetworkAttachmentDefinitionAvailable(clientset)
```

Comment:
we should fail only if ipam is enable and CRD doesnt exists, not in all the cases
which what happens here iiuc